### PR TITLE
Modernization, added new classes

### PR DESCRIPTION
**Pull request by BlindProrok**
Follow my lead and do not leave useless imports.

### CODE REVIEW

File: Bullet.py [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
-			raise NameError("Bullet type doesn't exist")
\ No newline at end of file
+
+
+class Bullet(DestructableObject):
+    def __init__(self, bultype, shooter, *args, **kwargs):
+        self.shooter = shooter
+        self.angle = self.shooter.angle
+        super().__init__(*args, **kwargs)
+        if bultype in Settings.Ammo:
```
Comment (ASIMER) [inline Bullet.py:11]: This type can contain rockets, etc.

File: Ship.py [hunk 2]
```diff
@@ -1,81 +1,11 @@
...
-			self.velocity = (self.velocity 
-							+ self.velocity.normalize().rotate(180) 
-							* 0.05)
-		else:
-			self.velocity = Vector(0, 0)
-	def overheat(self, heat_type):
-		self.heat[heat_type][0] += 0.006
-		if(self.heat[heat_type][0] >= 1):
-			self.heat[heat_type][0] = 1
-	def thrust(self, direction):
```
Comment (ASIMER) [inline Ship.py:47]: Need for another ships

File: Ship.py [hunk 3]
```diff
@@ -1,81 +1,11 @@
...
-		#(heat value, overheat(planned), thruster down timer(planned))
-		"forward_t": [0, 0, 0],
-		"backward_t": [0, 0, 0],
-		"left_t": [0, 0, 0],
-		"right_t": [0, 0, 0], 
-		"weapon_1": [0, 0, 0]
-		}
-	#inventory = Inventory()
-	source = 'img/usership.png'
-	def cooling(self, speed):
```
Comment (ASIMER) [inline Ship.py:31]: Need for another ships

File: Ship.py [hunk 4]
```diff
@@ -1,81 +1,11 @@
...
-from MovableObject import MovableObject
-from kivy.properties import ObjectProperty
-from kivy.vector import Vector
-from Bullet import Bullet
-from Settings import Settings
-class Ship(MovableObject):
-	hp = 0
-	mp = 0
-	xp = 0
-	mass = 1000000 #temporary
```
Comment (ASIMER) [inline Ship.py:14]: All vehicles have mass

File: UserShip.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+    def move(self):
+        self.coords = self.coords + self.velocity
+        for htype in self.heat:
+            if (self.heat[htype][0] > 0):
+                self.heat[htype][0] -= 0.004
+            else:
+                self.heat[htype][0] = 0
+
+    def fire(self):
```
Comment (ASIMER) [inline UserShip.py]: Ship method

File: UserShip.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+from Ship import Ship
+from kivy.vector import Vector
+from Bullet import Bullet
+from Settings import Settings
+from AbstractObjects import Singleton
+
+
+class UserShip(Ship, Singleton):
+    mass = 1000000
+    cooldown = {
```
Comment (ASIMER) [inline UserShip.py:8]: Ship dictionary

File: UserShip.py [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+        "Bullet": 0,
+    }
+    selected_ammo = "Bullet"
+    weapon = "Cannon"
```
Comment (ASIMER) [inline UserShip.py:12]: ship attribute

**Review by ASIMER — CHANGES_REQUESTED** [review]

File: AbstractObjects.py [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+from kivy.vector import Vector
+from kivy.properties import ObjectProperty
+
+
+class MovableObject(Image):
+
+    coords = ObjectProperty(Vector(0, 0), True)
+    velocity = ObjectProperty(Vector(0, 0), True)
+
+    def __init__(self, instance=None, mass=1000, max_speed=30, boost=1, max_boost=10, *args, **kwargs):
```
Comment (ASIMER) [inline AbstractObjects.py]: Background hasn't mass, "None" more practical

**Review by ASIMER — CHANGES_REQUESTED** [review]

File: AbstractObjects.py (hunk 8 above)
Comment (BlindProrok) [reply AbstractObjects.py]: Illya can fuck himself

File: UserShip.py (hunk 7 above)
Comment (BlindProrok) [reply UserShip.py:12]: pidr comment

File: UserShip.py (hunk 6 above)
Comment (BlindProrok) [reply UserShip.py:8]: huictionary

File: UserShip.py (hunk 5 above)
Comment (BlindProrok) [reply UserShip.py]: Agree. Sorry for the signature of the method. Only the fucking down could write THIS.

File: Ship.py (hunk 4 above)
Comment (BlindProrok) [reply Ship.py:14]: If they need

File: Ship.py (hunk 3 above)
Comment (BlindProrok) [reply Ship.py:31]: Need for another coders

File: Ship.py (hunk 2 above)
Comment (BlindProrok) [reply Ship.py:47]: Need for another coders

File: Bullet.py (hunk 1 above)
Comment (BlindProrok) [reply Bullet.py:11]: Code can contain logic

File: AbstractObjects.py (hunk 8 above)
Comment (ASIMER) [reply AbstractObjects.py]: чини сцуко

**Pull request merged**